compliance_checker/protocols/netcdf.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+'''
+compliance_checker/protocols/netcdf.py
+
+Functions to assist in determining if the URL points to a netCDF file
+'''
+
+
+def is_netcdf(url):
+    '''
+    Returns True if the URL points to a valid local netCDF file
+
+    :param str url: Location of file on the file system
+    '''
+    # Try an obvious exclusion of remote resources
+    if url.startswith('http'):
+        return False
+
+    # If it's a known extension, give it a shot
+    if url.endswith('nc'):
```

Comment:
If an empty or bad file has '.nc' it will crash when Dataset(ds_str) is executed, OR when an attribute of the dataset is accessed depending on the nature of the corruption.